### PR TITLE
Improve activity page monitoring

### PR DESCRIPTION
**Pull request by CatalinVoineag**
## Context

The previous [PR](https://github.com/DFE-Digital/apply-for-teacher-training/pull/10053) I did to monitor this page was not enough. This PR adds more information to be able to create a dashboard.

`activity_page_response` - response time
`activity_page_timeout` - if the page timed out, this is not an exact metric, it's based on previous response times that experienced time outs, over 115 seconds
`activity_page_providers` - the provider_ids, the activity log can show logs for multiple providers
 

## Changes proposed in this pull request

Added more columns to query in logit

## Guidance to review

I created a dummy dashboard using review data. Is there any other helpful information missing?

Screenshot from 2024-11-12 10-37-36

[Dashboard](https://kibana-uk1.logit.io/s/b2876053-0173-44fc-983d-99567292ba31/app/dashboards#/view/c41f3930-a0da-11ef-886b-455689d97b15?_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:now-2h,to:now))&_a=(description:'',filters:!(),fullScreenMode:!f,options:(hidePanelTitles:!f,useMargins:!t),query:(language:kuery,query:''),timeRestore:!f,title:'Provider%20Activity%20page',viewMode:view))

## Things to check

- [ ] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] If this code adds a column to the DB, decide whether it needs to be in analytics yml file or analytics blocklist
- [ ] API release notes have been updated if necessary
- [ ] If it adds a significant user-facing change, is it documented in the [CHANGELOG](CHANGELOG.md)?
- [ ] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)
- [ ] Inform data insights team due to database changes
- [x] Make sure all information from the Trello card is in here
- [x] Rebased main
- [x] Cleaned commit history
- [x] Tested by running locally
- [x] Add PR link to Trello card

### CODE REVIEW

**Comment by CatalinVoineag** [comment]
[Add benchmarking to Provider Activity Log](https://trello.com/c/nhuQ3EYD/414-add-benchmarking-to-provider-activity-log)

File: app/controllers/provider_interface/activity_log_controller.rb [hunk 1]
```diff
@@ -15,12 +15,15 @@ def index
...
       end
 
-      provider_ids = current_provider_user.providers.ids.join(', ')
-      if benchmark.real.between?(30, 115)
-        Rails.logger.info "Activity page slow for providers [#{provider_ids}] took #{benchmark.real.to_i} seconds"
-      end
-      if benchmark.real > 115
-        Rails.logger.info "Activity page timed out for providers [#{provider_ids}] took #{benchmark.real.to_i} seconds"
+      if benchmark.real > 30
+        provider_ids = current_provider_user.providers.ids.join(', ')
```
Comment (dcyoung-dev) [inline app/controllers/provider_interface/activity_log_controller.rb:19]: If we don't `join` these, would they get sent to Logit as an Array and allow us to filter the logs based on the array items?

File: app/controllers/provider_interface/activity_log_controller.rb [hunk 2]
```diff
@@ -15,12 +15,15 @@ def index
         end
       end
 
-      provider_ids = current_provider_user.providers.ids.join(', ')
-      if benchmark.real.between?(30, 115)
-        Rails.logger.info "Activity page slow for providers [#{provider_ids}] took #{benchmark.real.to_i} seconds"
-      end
-      if benchmark.real > 115
-        Rails.logger.info "Activity page timed out for providers [#{provider_ids}] took #{benchmark.real.to_i} seconds"
+      if benchmark.real > 30
```
Comment (dcyoung-dev) [inline app/controllers/provider_interface/activity_log_controller.rb:18]: How do we feel about 30seconds? To me it feels like a very long time - I'd class a page as slow if it took over 5secs to load - 10sec also seems fair https://www.nngroup.com/articles/website-response-times/#toc-response-time-limits-2

File: app/controllers/provider_interface/activity_log_controller.rb (hunk 1 above)
Comment (dcyoung-dev) [reply app/controllers/provider_interface/activity_log_controller.rb]: To note I think we should only store the ids in the array

File: app/controllers/provider_interface/activity_log_controller.rb (hunk 2 above)
Comment (CatalinVoineag) [reply app/controllers/provider_interface/activity_log_controller.rb:18]: I'll put 5

**Review by elceebee — APPROVED** [review]

**Review by dcyoung-dev — APPROVED** [review]
I like this, we may have to review after a week to make sure we're not spamming Logit - we have a limited allowance

**Pull request merged**